tableaudocumentapi/archivefile.py

```diff
@@ -0,0 +1,67 @@
+import contextlib
+import os
+import shutil
+import tempfile
+import zipfile
+
+import xml.etree.ElementTree as ET
+
+
+@contextlib.contextmanager
+def temporary_directory(*args, **kwargs):
+    d = tempfile.mkdtemp(*args, **kwargs)
+    try:
+        yield d
+    finally:
+        shutil.rmtree(d)
+
+
+def find_file_in_zip(zip, ext):
+    for filename in zip.namelist():
+        if os.path.splitext(filename)[-1].lower() == ext[:-1]:
```

Comment:
I prefer this because it's based on the actual content of the file rather than the filename/extension which I always find to be better by some definition of better.